serializable.ts

```diff
@@ -193,7 +193,7 @@ function fromJSON<T>(
           fromJSONStrategy = fromJSONDefault,
         } = serializablePropertyMap.getBySerializedKey(key) || {};
 
-        const processedValue: unknown = Array.isArray(value)
+        const processedValue: unknown = Array.isArray(value) && value.length
```

Comment:
@shardyMBAI Sorry, seems that this review comment didn't actually get added somehow

Adding in this constraint causes empty arrays to not be serialized into arrays:
```
test({
  name: "Revives an empty array of `type`",
  fn() {
    class OtherClass extends Serializable {
      @SerializeProperty()
      id!: number;
    }
    class Test extends Serializable {
      @SerializeProperty({
        fromJSONStrategy: (v) => new OtherClass().fromJSON(v),
      })
      array!: OtherClass[];
    }
    const testObj = new Test().fromJSON(
      `{"array":[]}`,
    );

    assert(Array.isArray(testObj.array)); // Test fails here - Test { array: OtherClass {} }
    assertEquals(testObj.array.length, 0);
  },
});
```

Reply:
damnit, thanks!